pkg/entrypoint/entrypointer.go

```diff
@@ -405,41 +405,43 @@ func replaceEnv(stepDir string) error {
 	for _, e := range os.Environ() {
 		pair := strings.SplitN(e, "=", 2)
 		matches := resultref.StepResultRegex.FindAllStringSubmatch(pair[1], -1)
+		v := pair[1]
 		for _, m := range matches {
 			replaceWith, _, err := findReplacement(stepDir, m[0])
 			if err != nil {
 				return err
 			}
-			os.Setenv(pair[0], strings.ReplaceAll(pair[1], m[0], replaceWith))
+			v = strings.ReplaceAll(v, m[0], replaceWith)
 		}
+		os.Setenv(pair[0], v)
 	}
 	return nil
 }
 
 // replaceCommandAndArgs performs replacements for step results in e.Command
 func replaceCommandAndArgs(command []string, stepDir string) ([]string, error) {
-	newCommand := []string{}
+	var newCommand []string
```

Comment:
NIT:
```suggestion
	var newCommands []string
```

Reply:
I think `command` is  supposed to be singular, although it's a slice of string, because it is a command which does one thing. `Commands` seems to indicate there are multiple commands to do many things. K8s spec for container also uses singular, although it's an a list of string.